back/libs/exceptions/src/filters/fc-web-html-exception.filter.ts

```diff
@@ -1,71 +1,155 @@
+import { ExceptionsConfig } from '@fc/exceptions/dto';
+import { ErrorPageParams } from '@fc/exceptions/types';
 import { LoggerService } from '@fc/logger';
 import { SessionService } from '@fc/session';
 
-import { generateErrorId } from '../helpers';
-import { FcBaseExceptionFilter } from './fc-base.exception-filter';
+import { BaseException } from '../exceptions/base.exception';
+import { CoreFcaBaseException } from '../exceptions/core-fca-base.exception';
+import { generateErrorId, getCode, getStackTraceArray } from '../helpers';
 
 @Catch(BaseException)
 @Injectable()
-export class FcWebHtmlExceptionFilter
-  extends FcBaseExceptionFilter
-  implements ExceptionFilter
-{
+export class FcWebHtmlExceptionFilter extends BaseExceptionFilter<BaseException> {
   constructor(
     protected readonly config: ConfigService,
     protected readonly session: SessionService,
     protected readonly logger: LoggerService,
   ) {
-    super(config, logger);
+    super();
   }
 
   catch(exception: BaseException, host: ArgumentsHost) {
-    const ctx = host.switchToHttp();
-    const res = ctx.getResponse();
+    const res = host.switchToHttp().getResponse();
 
     const code = this.getExceptionCodeFor(exception);
     const id = generateErrorId();
+    const staticMessage = exception.ui
+      ? messageDictionary[exception.ui] ||
+        messageDictionary['exceptions.default_message']
+      : undefined;
+    const message = exception.generic
+      ? exception.error_description
+      : staticMessage;
+
+    this.logException(code, id, message, exception);
 
-    const exceptionParam: ApiErrorParams = {
+    this.errorOutput({
+      error: { code, id, message },
       exception,
       res,
-      error: { code, id, message: exception.ui },
-      httpResponseCode: this.getHttpStatus(exception),
-      errorDetail: undefined,
+    });
+  }
+
+  protected getExceptionCodeFor(exception: BaseException): string {
+    const { prefix } = this.config.get<ExceptionsConfig>('Exceptions');
+
+    return exception.generic
+      ? exception.error
+      : getCode(exception.scope, exception.code, prefix);
+  }
+
+  protected logException<T extends BaseException>(
+    code: string,
+    id: string,
+    message: string,
+    exception: T,
+  ): void {
+    const exceptionObject = {
+      code,
+      id,
+      message,
+      originalError: exception.originalError,
+      reason: exception.log,
+      stackTrace: getStackTraceArray(exception),
+      type: exception.constructor.name,
+      statusCode: exception.http_status_code,
+    };
+
+    this.logger.error(exceptionObject);
+  }
+
+  protected errorOutput({
+    error,
+    exception,
+    res,
+  }: {
+    error: { code: string; id: string; message: string };
+    exception: BaseException;
+    res: Response;
+  }): void {
+    const errorPageParams: ErrorPageParams = {
+      error,
+      exceptionDisplay: {},
     };
 
-    this.logException(code, id, exception);
```

Comment:
Thought: I'm surprised to discover that this logException logic already existed before this PR. I don’t understand why we needed to explicitly add a log before an error occurs just to make sure we can see the result in Kibana, since this function seems to do the same thing 🤯